### PR TITLE
US-1047: Added Button component and cleaned up lint issues and more

### PR DESCRIPTION
**Pull request by jmiller-nexient**
Added the Button component but also cleaned up a bunch of lint errors and warnings I was getting after pulling the latest from master.

- .eslintrc - I added the './app' path to the resolver so all import paths will recognized relative to /app.
- removed a couple local LESS files that weren't adding anything not already defined in the default theme
- Added the return type for several functional components
- Fixed a few files that had 4 spaces instead of 2 for the indent size
- Moved all of the test/demo stuff for the custom components to the TestPage (instead of Home)
- Unsuccessfully tried to fix the failing test for Routes. This is something that needs to be resolved asap.

I also removed the snapshot tests on &lt;App> cuz I think it was skewing the coverage numbers. We'll add real tests when things like localization and, maybe, Redux, are added.

### CODE REVIEW

File: app/components/Button/tests/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React, { ReactElement } from 'react';
+import renderer from 'react-test-renderer';
+
+import Button from '..';
+
+describe('<Button />', () => {
+  it('should render correctly it is primary', () => {
```
Comment (JeremySparkman) [inline app/components/Button/tests/index.test.tsx:7]: more of a typo but missing when

File: app/components/Button/tests/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  it('should render correctly when it is primary and disabled', () => {
+    const tree: ReactElement = renderer.create(
+      <Button disabled type="primary" />,
+    ).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('should render correctly it is secondary', () => {
```
Comment (JeremySparkman) [inline app/components/Button/tests/index.test.tsx:23]: also missing when

File: app/components/Button/index.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    'dte-wismo-button-secondary': type === 'secondary',
+  });
+
+const Button: React.FC<IButtonProps> = (props: IButtonProps): React.ReactElement => {
+  const { children, className, disabled, htmlType, id, onClick, type } = props;
+
+  const classes: string = getClasses(className, type);
+
+  return (
+    // eslint-disable-next-line react/button-has-type
```
Comment (JeremySparkman) [inline app/components/Button/index.tsx:30]: Do we need to use // eslint-disable-next-line?

Comment (jmiller-nexient) [reply app/components/Button/index.tsx:30]: Yea, there's a funny rule about setting the type attribute when using a variable (htmlType). I forget the details but there's a lot of discussion about it online.

File: vim.exe.stackdump [hunk 4]
```diff
@@ -0,0 +1,19 @@
+Stack trace:
```
Comment (JeremySparkman) [inline vim.exe.stackdump:1]: I think this was added on accident

**Pull request merged**